### PR TITLE
Make Shanghai the default EVM version.

### PR DESCRIPTION
**Pull request by ekpyron**
Given our discussion with tooling, we'll flip the switch and will prominently note this in the release notes.

Has hardhat fail hard so far, need to check if we can bump some versions somewhere, supposedly the latest release has support.

Part of https://github.com/ethereum/solidity/issues/14073

### CODE REVIEW

**Comment by axic** [comment]
I'd wait for the evmone release, because the one used likely still has EOF enabled on Shanghai.

**Comment by clabby** [comment]
Not that this should block this PR by any means, but just as a note - some tooling is still catching up to Shanghai, namely [Foundry](https://github.com/foundry-rs/foundry) (although, @evalir is [working on upgrading to REVM >=3.0](https://github.com/foundry-rs/foundry/pull/4757), which will allow foundry to support Shanghai with relative ease).

**Comment by ekpyron** [comment]
> Not that this should block this PR by any means, but just as a note - some tooling is still catching up to Shanghai, namely [Foundry](https://github.com/foundry-rs/foundry) (although, @Evalir is [working on upgrading to REVM >=3.0](https://github.com/foundry-rs/foundry/pull/4757), which will allow foundry to support Shanghai with relative ease).

Yes, we're aware and brought this up in our communication channels with tooling - hardhat didn't think this would be an issues for them (even though our CI here looks otherwise, so we'll have to look into that) and we were told that foundry always inserts an explicit evm version, so they won't be affected by us changing the default (and can add support and flip the evm version they pass on to us in their own time).

**Comment by ekpyron** [comment]
> I'd wait for the evmone release, because the one used likely still has EOF enabled on Shanghai.

Well, that doesn't really affect us significantly, though, does it? And is independent of making this the default - that also already affects the plain support. We plan to release at least the support soon - this PR, i.e. switching to default, we can still postpone for one more release, but I had planned to also see if we can get it in.

**Comment by axic** [comment]
> Well, that doesn't really affect us significantly, though, does it? And is independent of making this the default - that also already affects the plain support. 

It affects both, and ideally we get confirmation with a proper evmone release before this goes live. Evmone likely will be released by next week.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1619,17 +1619,19 @@ workflows:
       - t_ext: *job_native_test_ext_gnosis
       - t_ext: *job_native_test_ext_zeppelin
       - t_ext: *job_native_test_ext_ens
-      - t_ext: *job_native_test_ext_trident
-      - t_ext: *job_native_test_ext_euler
+      # NOTE: The external tests below were commented because they
+      # depend on a specific version of hardhat which does not support shanghai EVM.
```
Comment (ekpyron) [inline .circleci/config.yml]: This should either be above *all* of the commented test runs or you move the others directly here. The idea of the comment is to see for which ones we did this without disecting git comments :-).
Collecting all of them together here is easiest.
The benchmark run should then also get such a comment with all of the runs there collected below it.

**Comment by axic** [comment]
> I'd wait for the evmone release, because the one used likely still has EOF enabled on Shanghai.

Release coming out today or tomorrow.

**Comment by ekpyron** [comment]
> > I'd wait for the evmone release, because the one used likely still has EOF enabled on Shanghai.
> 
> Release coming out today or tomorrow.

Alright - I'll be merging this today and we can bump the evmone version as soon as it's available.
We plan to release on Wednesday - it would  be nice if we had evmone bumped by then - otherwise we'll release anyways, since the likelihood of actually severe test mismatches resulting from using the current evmone is very low.

**Review by r0qs — APPROVED** [review]

**Comment by axic** [comment]
> Release coming out today or tomorrow.

 https://github.com/ethereum/evmone/releases/tag/v0.10.0

Let's hope all the tests pass, otherwise this will be the most yolo Solidity release so far 😅

**Comment by ekpyron** [comment]
> > Release coming out today or tomorrow.
> 
> https://github.com/ethereum/evmone/releases/tag/v0.10.0
> 
> Let's hope all the tests pass, otherwise this will be the most yolo Solidity release so far sweat_smile

The only substantial codegen change for shanghai is push0, and that was already supported by the old evmone release - the danger really wasn't so big :-D. And yeah, all tests still pass in https://github.com/ethereum/solidity/pull/14200 except slight gas accounting differences in contract creation, so it *was* fine :-).

**Comment by axic** [comment]
> The only substantial codegen change for shanghai is push0, and that was already supported by the old evmone release - the danger really wasn't so big :-D. 

The problem wasn't that, but that the evmone being used had a ton of EOF stuff enabled for Shanghai.

**Comment by ekpyron** [comment]
> > The only substantial codegen change for shanghai is push0, and that was already supported by the old evmone release - the danger really wasn't so big :-D.
> 
> The problem wasn't that, but that the evmone being used had a ton of EOF stuff enabled for Shanghai.

Yes, but it's not like we generate anything that'd have been affected by that. But yeah, the version is bumped now, so it's fine.

**Comment by axic** [comment]
I was more worried that something unintentionally get accepted by that version of evmone 😀

**Pull request merged**